src/ophys_etl/modules/segmentation/graph_utils/feature_vector_rois.py

```diff
@@ -0,0 +1,522 @@
+
+        filter_fraction: float
+            Fraction of brightest timesteps to use in calculating
+            features.
+
+        pixel_ignore: Optional[np.ndarray]
+            A (n_rows, n_cols) array of booleans that is True
+            for any pixel to be ignored, presumably because it
+            has already been assigned to an ROI (default: None)
+
+        rng: Optional[np.random.RandomState]
+            A random number generator (used by
+            calculate_pearson_feature_vectors to select pixels
+            for use in choosing brightest timesteps)
+
+        Returns
+        -------
+        features: np.ndarray
+            A (n_pixels, n_features) array in which each row is the
+            feature vector corresponding to a pixel in sub_video.
+
+        Notes
+        ------
+        features[ii, :] can be mapped into a pixel in sub_video using
+        self.index_to_pixel[ii]
+        """
+        msg = "PotentialROI does not implement get_features; "
+        msg += "must specify a valid sub-class"
+        raise NotImplementedError(msg)
+
+    def calculate_feature_distances(
+            self,
+            sub_video: np.ndarray,
+            filter_fraction: float,
+            pixel_ignore: Optional[np.ndarray] = None,
+            rng: Optional[np.random.RandomState] = None):
+        """
+        Set self.feature_distances, an (n_pixel, n_pixel) array
+        of distances between pixels in feature space.
+
+        Parameters
+        ----------
+        sub_video: np.ndarray
+            The subset of video data being scanned for an ROI.
+            Shape is (n_time, n_rows, n_cols)
+
+        filter_fraction: float
+            Fraction of brightest timesteps to use in calculating
+            features.
+
+        pixel_ignore: Optional[np.ndarray]
+            A (n_rows, n_cols) array of booleans that is True
+            for any pixel to be ignored, presumably because it
+            has already been assigned to an ROI (default: None)
+
+        rng: Optional[np.random.RandomState]
+            A random number generator (used by
+            calculate_pearson_feature_vectors to select pixels
+            for use in choosing brightest timesteps)
+
+        Notes
+        ------
+        self.feature_distances[ii, jj] can be mapped into pixel coordinates
+        using self.index_to_pixel[ii], self.index_to_pixel[jj]
+        """
+
+        features = self.get_features(sub_video,
+                                     filter_fraction,
+                                     pixel_ignore=pixel_ignore,
+                                     rng=rng)
+
+        self.feature_distances = cdist(features,
+                                       features,
+                                       metric='euclidean')
+
+        self.roi_mask = np.zeros(self.n_pixels, dtype=bool)
+        self.roi_mask[self.pixel_to_index[self.seed_pt]] = True
+
+    def get_not_roi_mask(self) -> None:
```

Comment:
this would be more clear and easier to test if it was more like:
```
def get_not_roi_mask(roi_mask, feature_distances):
    ...
    return roi_mask
```
and was a stand-alone function or static method or something

Reply:
Done